core/src/test/java/io/questdb/mp/SimpleLockTest.java

```diff
@@ -0,0 +1,243 @@
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.mp;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.locks.LockSupport;
+
+import static junit.framework.Assert.assertFalse;
+
+public class SimpleLockTest {
+    @Test
+    public void testLock() {
+        SimpleWaitingLock lock = new SimpleWaitingLock();
+        Assert.assertFalse(lock.isLocked());
+        lock.lock();
+        Assert.assertTrue(lock.isLocked());
+        lock.unlock();
+        Assert.assertFalse(lock.isLocked());
+
+        Assert.assertTrue(lock.tryLock());
+        Assert.assertFalse(lock.tryLock());
+
+        lock.unlock();
+        Assert.assertTrue(lock.tryLock(10, TimeUnit.MILLISECONDS));
+        Assert.assertFalse(lock.tryLock(10, TimeUnit.MILLISECONDS));
+        Assert.assertTrue(lock.isLocked());
+
+        try {
+            lock.unlock();
+            lock.unlock();
+            Assert.fail();
+        } catch (IllegalStateException ex) {
+            Assert.assertFalse(lock.isLocked());
+        }
+    }
+
+    @Test
+    public void testLock2() throws Exception {
+        SimpleWaitingLock lock = new SimpleWaitingLock();
+        AtomicBoolean holdsLock = new AtomicBoolean();
+
+        Thread thread;
+        lock.lock();
+        try {
+            thread = new Thread(() -> {
+                lock.lock();  // should block
+                holdsLock.set(true);
+                LockSupport.park();
+                lock.unlock();
+                holdsLock.set(false);
+            });
+            thread.start();
+
+            // give time for thread to block
+            Thread.sleep(500);
+            Assert.assertFalse(holdsLock.get());
+        } finally {
+            lock.unlock();
+        }
+
+        // thread should acquire lock, park, unpark, and then release lock
+        while (!holdsLock.get()) {
+            Thread.sleep(20);
+        }
+        LockSupport.unpark(thread);
+        while (holdsLock.get()) {
+            Thread.sleep(20);
+        }
+        thread.join();
+    }
+
+    @Test
+    public void testLock3() throws Exception {
+        SimpleWaitingLock lock = new SimpleWaitingLock();
+        Thread thread = new Thread(() -> {
+            lock.lock();
+            try {
+                LockSupport.park();
+            } finally {
+                lock.unlock();
+            }
+        });
+        thread.start();
+
+        // wat for thread to acquire lock
```

Comment:
nit: the same typo is also in the further tests
```suggestion
        // wait for thread to acquire lock
```